src/raftstore/store/peer.rs

```diff
@@ -1657,9 +1666,9 @@ impl Peer {
             responses.push(resp);
         }
 
-        let mut resp = RaftCmdResponse::new();
-        resp.set_responses(protobuf::RepeatedField::from_vec(responses));
-        Ok(resp)
+        let mut response = RaftCmdResponse::new();
+        response.set_responses(protobuf::RepeatedField::from_vec(responses));
+        Ok(ReadResponse { response, snapshot })
```

Comment:
If there is no snap request in requests, could we use `None` as `ReadResponse::snapshot` in order to release `snapshot` as soon as possible?